frame/contracts/src/wasm/mod.rs

```diff
@@ -41,81 +37,63 @@ pub use crate::wasm::{
+	MemoryType, Module, StackLimits, Store,
 };
+const BYTES_PER_PAGE: usize = 64 * 1024;
 
-/// A prepared wasm module ready for execution.
-///
-/// # Note
-///
-/// This data structure is mostly immutable once created and stored. The exceptions that
-/// can be changed by calling a contract are `instruction_weights_version` and `code`.
-/// `instruction_weights_version` and `code` change when a contract with an outdated instrumentation
-/// is called. Therefore one must be careful when holding any in-memory representation of this
-/// type while calling into a contract as those fields can get out of date.
-#[derive(Clone, Encode, Decode, scale_info::TypeInfo, MaxEncodedLen)]
+/// Validated Wasm module ready for execution.
+/// This data structure is immutable once created and stored.
+#[derive(Encode, Decode, scale_info::TypeInfo)]
 #[codec(mel_bound())]
 #[scale_info(skip_type_params(T))]
-pub struct PrefabWasmModule<T: Config> {
-	/// Version of the instruction weights with which the code was instrumented.
-	#[codec(compact)]
-	instruction_weights_version: u32,
-	/// Initial memory size of a contract's sandbox.
-	#[codec(compact)]
-	initial: u32,
-	/// The maximum memory size of a contract's sandbox.
-	#[codec(compact)]
-	maximum: u32,
-	/// Code instrumented with the latest schedule.
-	code: RelaxedCodeVec<T>,
-	/// A code that might contain non deterministic features and is therefore never allowed
-	/// to be run on chain. Specifically this code can never be instantiated into a contract
-	/// and can just be used through a delegate call.
-	determinism: Determinism,
-	/// The uninstrumented, pristine version of the code.
-	///
-	/// It is not stored because the pristine code has its own storage item. The value
-	/// is only `Some` when this module was created from an `original_code` and `None` if
-	/// it was loaded from storage.
-	#[codec(skip)]
-	original_code: Option<CodeVec<T>>,
-	/// The code hash of the stored code which is defined as the hash over the `original_code`.
-	///
-	/// As the map key there is no need to store the hash in the value, too. It is set manually
-	/// when loading the module from storage.
+pub struct WasmBlob<T: Config> {
+	code: CodeVec<T>,
+	// This isn't needed for contract execution and is not stored alongside it.
 	#[codec(skip)]
-	code_hash: CodeHash<T>,
-	// This isn't needed for contract execution and does not get loaded from storage by default.
-	// It is `Some` if and only if this struct was generated from code.
-	#[codec(skip)]
-	owner_info: Option<OwnerInfo<T>>,
+	pub code_info: CodeInfo<T>,
 }
 
-/// Information that belongs to a [`PrefabWasmModule`] but is stored separately.
+/// Contract code related data, such as:
+///
+/// - owner of the contract, i.e. account uploaded its code,
+/// - storage deposit amount,
+/// - reference count,
+/// - determinism marker.
 ///
 /// It is stored in a separate storage entry to avoid loading the code when not necessary.
```

Comment:
this comment is not really true anymore

Reply:
> this comment is not really true anymore

what exactly is not true in it? we still don't need to load the code if just need its metadata